eventmesh-connectors/eventmesh-connector-http/src/main/java/org/apache/eventmesh/connector/http/sink/handle/CommonHttpSinkHandler.java

```diff
@@ -0,0 +1,197 @@
+            .onSuccess(res -> {
+                Long timestamp = record.getTimestamp();
+                Map<String, ?> offset = record.getPosition().getOffset().getOffset();
+                log.info("Request sent successfully. Record: timestamp={}, offset={}", timestamp, offset);
+                // Determine whether the status code is 200
+                if (res.statusCode() == HttpResponseStatus.OK.code()) {
+                    // store the received data, when webhook is enabled
+                    if (this.connectorConfig.isWebhook()) {
+                        String dataStr = res.body().toString();
+                        if (dataStr.isEmpty()) {
+                            log.warn("Received data is empty.");
+                            return;
+                        }
+                        JSONObject receivedData = JSON.parseObject(dataStr);
+                        if (receivedDataQueue.size() == Integer.MAX_VALUE) {
+                            // if the queue is full, remove the oldest element
+                            JSONObject removedData = receivedDataQueue.poll();
+                            log.info("The queue is full, remove the oldest element: {}", removedData);
+                        }
+                        boolean b = receivedDataQueue.offer(receivedData);
+                        if (b) {
+                            log.info("Successfully put the received data into the queue: {}", receivedData);
+                        } else {
+                            log.error("Failed to put the received data into the queue: {}", receivedData);
+                        }
+                    }
+                } else {
+                    log.error("Unexpected response received. Record: timestamp={}, offset={}. Response: code={} header={}, body={}",
+                        timestamp,
+                        offset,
+                        res.statusCode(),
+                        res.headers(),
+                        res.body().toString()
+                    );
+                }
+            })
+            .onFailure(err -> {
+                Long timestamp = record.getTimestamp();
+                Map<String, ?> offset = record.getPosition().getOffset().getOffset();
+                log.error("Request failed to send. Record: timestamp={}, offset={}", timestamp, offset, err);
+            });
+    }
+
+    /**
+     * Convert ConnectRecord to HttpConnectRecord
+     *
+     * @param record the ConnectRecord to convert
+     * @return the converted HttpConnectRecord
+     */
+    private HttpConnectRecord convertToHttpConnectRecord(ConnectRecord record) {
+        HttpConnectRecord httpConnectRecord = new HttpConnectRecord();
+        httpConnectRecord.setType(this.type);
+        LocalDateTime currentTime = LocalDateTime.now();
+        httpConnectRecord.setTimestamp(currentTime.toString());
+        httpConnectRecord.setData(record);
+        return httpConnectRecord;
+    }
```

Comment:
You can use a constructer instead of setters, because constructer has a better performance when instantiating an object.

Reply:
Resolved at https://github.com/apache/eventmesh/pull/4837/commits/fe2b7327b744fcf0f3ea3576c892c551ed1d6fb4